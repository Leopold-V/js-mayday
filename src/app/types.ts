export type userType = {
  userId: string;
  username: string;
  email: string;
  avatar: string;
  bio: string;
};

export type questionType = {
  id: string;
  title: string;
  authorId: string;
  content: string;
  date: string;
};
