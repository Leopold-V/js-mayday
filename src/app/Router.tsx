import React from 'react';
import { useSelector } from 'react-redux';
import { Redirect, Route, Switch } from 'react-router-dom';

import {
  Home,
  Dashboard,
  ReadingList,
  Faq,
  About,
  Tags,
  NotFound,
  LoginPage,
  SignupPage,
  SettingsPage,
  ProfilePage,
  CreateQuestionPage,
  QuestionPage,
  QuestionEditPage,
} from '../pages';
import { SearchPage } from '../pages/SearchPage';
import { TagsPage } from '../pages/TagsPage';

export const Router = () => {
  const loading = useSelector((state: any) => state.user.loading);
  const isAuth = useSelector((state: any) => state.user.isAuthenticated);

  return (
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/question/:id" component={QuestionPage} />
      <Route exact path="/profile/:id" component={ProfilePage} />
      <Route exact path="/faq" component={Faq} />
      <Route exact path="/about" component={About} />
      <Route exact path="/search/:input" component={SearchPage} />
      <Route
        exact
        path="/dashboard"
        render={() => !loading && (isAuth ? <Dashboard /> : <Redirect to="/login" />)}
      />
      <Route
        exact
        path="/readinglist"
        render={() => !loading && (isAuth ? <ReadingList /> : <Redirect to="/login" />)}
      />
      <Route
        exact
        path="/tags/:tagname"
        render={(props) => !loading && (isAuth ? <Tags {...props} /> : <Redirect to="/login" />)}
      />
      <Route
        exact
        path="/login"
        render={() => !loading && (!isAuth ? <LoginPage /> : <Redirect to="/" />)}
      />
      <Route
        exact
        path="/signup"
        render={() => !loading && (!isAuth ? <SignupPage /> : <Redirect to="/" />)}
      />
      <Route
        exact
        path="/settings"
        render={() => !loading && (isAuth ? <SettingsPage /> : <Redirect to="/login" />)}
      />
      <Route
        exact
        path="/new"
        render={() => !loading && (isAuth ? <CreateQuestionPage /> : <Redirect to="/login" />)}
      />
      <Route
        exact
        path="/edit/:id"
        render={(props) =>
          !loading && (isAuth ? <QuestionEditPage {...props} /> : <Redirect to="/login" />)
        }
      />
      <Route
        exact
        path="/tags"
        render={() => !loading && (isAuth ? <TagsPage /> : <Redirect to="/login" />)}
      />
      <Route component={NotFound} />
    </Switch>
  );
};
