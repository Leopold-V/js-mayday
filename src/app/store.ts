import { configureStore } from '@reduxjs/toolkit';
import userReducer from '../slices/userSlice';
import questionReducer from '../slices/questionSlice';

const store = configureStore({
  reducer: {
    questions: questionReducer,
    user: userReducer,
  },
});

export default store;
