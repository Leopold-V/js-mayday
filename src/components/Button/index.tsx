export { ButtonCreateQuestion } from './ButtonCreateQuestion';
export { ButtonLike } from './ButtonLike';
export { ButtonLogin } from './ButtonLogin';
export { ButtonProfile } from './ButtonProfile';
export { ButtonNotif } from './ButtonNotif';
export { ButtonRead } from './ButtonRead';
export { ButtonSignup } from './ButtonSignup';
