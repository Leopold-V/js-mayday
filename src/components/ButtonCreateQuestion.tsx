import React from 'react';
import { Link } from 'react-router-dom';

export const ButtonCreateQuestion = () => {
  return (
    <Link to="/new" className="btn-primary text-gray font-light">
      New question
    </Link>
  );
};
