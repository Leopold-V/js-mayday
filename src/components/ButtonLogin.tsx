import React from 'react'
import { useHistory } from 'react-router'


export const ButtonLogin = () => {
    let history = useHistory();

    const handleClick = () => {
        history.push('/login');
    }

    return (
        <button onClick={handleClick} className="btn-primary text-gray font-light">
           Login 
        </button>
    )
}
