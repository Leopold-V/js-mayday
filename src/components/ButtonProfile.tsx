import React from 'react'

export const ButtonProfile = () => {
    const handleClick = () => {
        console.log('Display profile menu');
    }

    return (
        <button onClick={handleClick}>
            <svg width="30" height="30">
                <circle 
                    fill='#ffffff'
                    r="15"
                    cx="15"
                    cy="15"
                />
            </svg>
        </button>
    )
}
