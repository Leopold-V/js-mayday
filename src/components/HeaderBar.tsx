import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';

import { ButtonProfile, ButtonCreateQuestion, ButtonLogin, ButtonSignup, ButtonNotif} from './Button';
import { SearchBar } from './SearchBar';
import { ProfileDropwdown } from './ProfileDropwdown';

export const HeaderBar = () => {
  const user = useSelector((state: any) => state.user.user);
  const loading = useSelector((state: any) => state.user.loading);
  const [showProfileItems, setShowProfileItems] = useState(false);

  return (
    <div className="bg-gray sticky top-0 z-50">
      <nav className="h-14 w-11/12 lg:w-10/12 py-2 mx-auto flex justify-between">
        <div className="flex items-center space-x-4">
          <div className="bg-white hover:opacity-60 text-secondary rounded flex items-center px-1 justify-center h-10 transition duration-300">
            <Link to="/" className="font-bold text-lg text-gray">
              Mdev
            </Link>
          </div>
          <SearchBar />
        </div>
        {!loading &&
          (user ? (
            <div className="flex items-center space-x-4">
              <ButtonCreateQuestion />
              <ButtonNotif />
              <ButtonProfile setShow={setShowProfileItems} />
              <ProfileDropwdown show={showProfileItems} setShow={setShowProfileItems} />
            </div>
          ) : (
            <div className="flex items-center space-x-4">
              <ButtonLogin />
              <ButtonSignup />
            </div>
          ))}
      </nav>
    </div>
  );
};
