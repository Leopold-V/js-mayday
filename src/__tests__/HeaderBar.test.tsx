import React from 'react';
import { render, screen } from '@testing-library/react';
import { HeaderBar } from '../components/HeaderBar';
import { notificationType, userType } from '../app/types';
import * as redux from 'react-redux'
import { BrowserRouter } from 'react-router-dom';

const spy = jest.spyOn(redux, 'useSelector')
spy.mockReturnValue(null)

describe('Headerbar works correctly', () => {
  const user: userType = {
    avatar: "",
    bio: "Cool",
    dev_profile: "",
    email: "leopold12d12@gmail.com",
    location: "France",
    score: 2,
    tags: [],
    userId: "7Rh5lL56AHMFRlZ3pEGnLcxBlUl2",
    username: "Joe",
    website_url: "https://dev.to/leopold" 
};
  const loading = false;
  const notifications: notificationType[] = [];

  const useSelector = jest.fn();

  it('renders without crashing', () => {
    render(<BrowserRouter><HeaderBar /></BrowserRouter>);
    const text = screen.getByText('Mdev');
    expect(text).toBeInTheDocument();
  });
})
