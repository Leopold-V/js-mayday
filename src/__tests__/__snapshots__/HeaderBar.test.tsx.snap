// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headerbar works correctly renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-gray sticky top-0 z-50"
      >
        <nav
          class="h-14 w-11/12 lg:w-10/12 py-2 mx-auto flex justify-between"
        >
          <div
            class="flex items-center space-x-4"
          >
            <div
              class="bg-white hover:opacity-60 text-secondary rounded flex items-center px-1 justify-center h-10 transition duration-300"
            >
              <a
                class="font-bold text-lg text-gray"
                href="/"
              >
                Mdev
              </a>
            </div>
            <form
              class="hidden md:block"
            >
              <input
                class="border shadow-lg border-gray-600 rounded py-2 px-2 bg-gray-100 focus:outline-none focus:ring-1 focus:ring-primary focus:bg-background focus:border-transparent h-10 w-96 placeholder-gray-500 transition duration-200"
                placeholder="Search..."
                value=""
              />
            </form>
          </div>
          <div
            class="flex items-center space-x-4"
          >
            <a
              class="btn-primary text-gray"
              href="/login"
            >
              Login
            </a>
            <a
              class="btn-white text-gray font-light"
              href="/signup"
            >
              Signup
            </a>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-gray sticky top-0 z-50"
    >
      <nav
        class="h-14 w-11/12 lg:w-10/12 py-2 mx-auto flex justify-between"
      >
        <div
          class="flex items-center space-x-4"
        >
          <div
            class="bg-white hover:opacity-60 text-secondary rounded flex items-center px-1 justify-center h-10 transition duration-300"
          >
            <a
              class="font-bold text-lg text-gray"
              href="/"
            >
              Mdev
            </a>
          </div>
          <form
            class="hidden md:block"
          >
            <input
              class="border shadow-lg border-gray-600 rounded py-2 px-2 bg-gray-100 focus:outline-none focus:ring-1 focus:ring-primary focus:bg-background focus:border-transparent h-10 w-96 placeholder-gray-500 transition duration-200"
              placeholder="Search..."
              value=""
            />
          </form>
        </div>
        <div
          class="flex items-center space-x-4"
        >
          <a
            class="btn-primary text-gray"
            href="/login"
          >
            Login
          </a>
          <a
            class="btn-white text-gray font-light"
            href="/signup"
          >
            Signup
          </a>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
